### ISSUE TITLE
Algorithm SAX should incorporate rescaling of the data.

### ISSUE THREAD

**Issue by ChristoferNal** [tslearn-team/tslearn#237]
**Describe the bug**
Algorithm SAX needs the data to be scaled to a normal distribution, otherwise it doesn't work correctly. 

This is not stated clearly in the documentation (https://tslearn.readthedocs.io/en/stable/gen_modules/piecewise/tslearn.piecewise.SymbolicAggregateApproximation.html#tslearn.piecewise.SymbolicAggregateApproximation) and may lead to misuse of the library. 

I would suggest to incorporate the scaling within the algorithm and having a flag defaulting to scale=True. 

**To Reproduce**
According to the documentation, someone would use the algorithm as follows:
sax = SymbolicAggregateApproximation(n_segments=n_paa_segments,
                                     alphabet_size_avg=n_sax_symbols)
sax_dataset_inv = sax.inverse_transform(sax.fit_transform(dataset))
which is wrong because segments' edges should be quantiles from a normal distribution

**Expected behavior**
There is only an example that does the scaling https://tslearn.readthedocs.io/en/stable/auto_examples/misc/plot_sax.html#sphx-glr-auto-examples-misc-plot-sax-py and thus SAX works as expected.

**Environment (please complete the following information):**
 - OS: [independent]
 - tslearn version [0.3.1]

**Additional context**
A quick way to avoid misusing the library is to update the documentation that describes the algorithm.
The best way would be to include the step of converting the given time series to a normal distribution withing the algorithm, as this is a necessary step for SAX but not for other algorithms.

**Comment by GillesVandewiele** [comment]
Hi @ChristoferNal,

Thank you for this very clear issue description! 

We indeed had a discussion about this in #152. It seems like we agreed on always standard-normalizing the input provided to a KNeighbors algorithm when 'sax' is used as a metric, but that I forgot to actually implement this... But as you point out, this normalization should perhaps happen within the SAX transformation class itself...

In terms of the examples, it seems like one of the examples is not shown in the gallery of examples (https://github.com/tslearn-team/tslearn/blob/master/tslearn/docs/examples/neighbors/plot_sax_mindist_knn.py). @rtavenar, any idea what the cause of that could be?

**Comment by rtavenar** [comment]
@GillesVandewiele it is present in the `latest` version of the docs, just not there yet in the `stable` one, so it will be available soon (when we release v0.4).

@ChristoferNal Your option seems to be the most reasonable one (implementing this in SAX directly as an extra parameter). For some time (like from v0.4 to 0.6 or so), we may have to trigger a warning that this is done by default and if one wants to discard it, it should be discarded explicitly. Would you be willing to implement a fix for that?

**Comment by ChristoferNal** [comment]
Do you mean that there will be a warning to scale the data manually before passing them to SAX? I can fix it and incorporate the scaling inside SAX for later versions.

**Comment by rtavenar** [comment]
No I mean we should have a scale parameter that would default to False yet raise a warning that the default will change in a future version.

**Comment by rtavenar** [comment]
@ChristoferNal if you were able to open a PR that would implement that, I'd be glad to review it.

**Comment by ChristoferNal** [comment]
@rtavenar As I can see you already fixed the issue. Apologies for the late reply, but I have been very busy.
